distributed/utils.py

```diff
@@ -941,9 +941,11 @@ def truncate_exception(e, n=10000):
 
 def validate_key(k):
     """Validate a key as received on a stream."""
-    typ = type(k)
-    if typ is not str and typ is not bytes:
-        raise TypeError(f"Unexpected key type {typ} (value: {k!r})")
+    if isinstance(k, tuple):
+        for e in k:
+            validate_key(e)
```

Comment:
I'd rather disallow nested tuples

Reply:
Also this function feels like it should be in dask/dask.

I'm aware that dask/dask can run on top of schedulers other than dask/distributed, but I'd much rather not have a tighter restriction on dask/distributed than in dask/dask.